packages/design/src/DataTable/useTable.ts

```diff
@@ -172,14 +173,39 @@ function searchAndFilterCb<T>(
   searchValue: string,
   propName: keyof T & string
 ) {
-  if (propName === 'tags') {
-    return targetValue.some(item => {
-      return item.toLocaleLowerCase().includes(searchValue.toLocaleLowerCase());
-    });
-  }
   if (propName.toLocaleLowerCase().includes('date')) {
     return displayDate(targetValue).includes(searchValue);
   }
+  if (propName.toLocaleLowerCase().includes('time')) {
+    return displayDateTime(targetValue).includes(searchValue);
+  }
+  if (typeof targetValue === 'boolean') {
+    return (
+      propName.toLocaleLowerCase().includes(searchValue.toLocaleLowerCase()) &&
+      targetValue
+    );
+  }
+
+  // For searching through elements of an array
+  if (Array.isArray(targetValue)) {
```

Comment:
this assumes the items are of type `string`, what happens if its other than string?